tests/Doctrine/Tests/Common/Cache/FileCacheTest.php

```diff
@@ -83,36 +84,29 @@ public function testFilenameCollision()
 
         foreach ($data as $key => $expected) {
             $path   = $method->invoke($cache, $key);
-            $actual = pathinfo($path, PATHINFO_FILENAME);
 
             $this->assertNotContains($path, $paths);
-            $this->assertEquals($expected, $actual);
 
             $paths[] = $path;
         }
     }
 
-    public function testFilenameShouldCreateThePathWithFourSubDirectories()
+    public function testFilenameShouldCreateThePathWithOneSubDirectory()
     {
         $cache          = $this->driver;
         $method         = new \ReflectionMethod($cache, 'getFilename');
         $key            = 'item-key';
         $expectedDir    = array(
-            '84', 'e0', 'e2', 'e8', '93', 'fe', 'bb', '73', '7a', '0f', 'ee',
-            '0c', '89', 'd5', '3f', '4b', 'b7', 'fc', 'b4', '4c', '57', 'cd',
-            'f3', 'd3', '2c', 'e7', '36', '3f', '5d', '59', '77', '60'
+            '84',
         );
         $expectedDir    = implode(DIRECTORY_SEPARATOR, $expectedDir);
 
         $method->setAccessible(true);
 
         $path       = $method->invoke($cache, $key);
-        $filename   = pathinfo($path, PATHINFO_FILENAME);
         $dirname    = pathinfo($path, PATHINFO_DIRNAME);
 
-        $this->assertEquals('item__key', $filename);
```

Comment:
These tests dig deep into private/protected API to test specific behaviors. If you are reducing the units being tested, then there should be one test that covers the functional behavior (still an unit test, but integrates both these tested units)